runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts

```diff
@@ -106,12 +120,62 @@ export default class ExcelActionHandler extends StatefulActionHandler {
         cell.Value = input.value;
     }
 
+    async setCells(
+        input: ExcelSetCellsActionInput
+    ): Promise<void> {
+        const { worksheet: optionalWorksheet, targetExcelStructure } = input;
+        const openedWorksheet = this.session.ActiveSheet.name;
+        const isObject = this.checkIsObject(targetExcelStructure), isArray = this.checkIsArray(targetExcelStructure);
+
+        if (optionalWorksheet) this.session.Worksheets(optionalWorksheet).Activate();
```

Comment:
Maybe we should first check if we're already on the optional worksheet before we start switching?